src/policy/offload.rs

```diff
@@ -212,4 +221,142 @@ impl OffloadPolicy {
     unsafe fn set_physical_ssip(&self) {
         Arch::set_csr_bits(Csr::Mip, mie::SSIE_FILTER);
     }
+
+    /// This function is a rust implementation of the function "sbi_trap_redirect" in the sbi_trap.c from the OpenSBI codebase
+    /// This function corresponds to a payload to payload transition. Therefore, we don't modify the virtual context but the physical registers
+    /// The only exception is ctx.pc because in the trap handler we write mepc with ctx.pc value.
+    #[allow(unused)]
+    fn redirect_trap_to_s_mode(ctx: &mut VirtContext) -> PolicyHookResult {
+        let mut mstatus = ctx.trap_info.mstatus;
+
+        // The previous virtualisation mode
+        let prev_virt: bool = mstatus & MPV_FILTER != 0;
```

Comment:
Small nit:

For booleans a good naming convention is to have a `is` somewhere, such as `is_prev_virt` or `prev_is_virt`. That makes it easier to identify boolean variables, and also makes conditionals more readable, like `is prev_is_virt {}`.